src/RedHat.AspNetCore.Server.Kestrel.Transport.Linux/Transport.cs

```diff
@@ -73,50 +70,44 @@ public async Task BindAsync()
-                            File.Delete(_endPoint.SocketPath);
-                            socket.Bind(_endPoint.SocketPath);
-                            socket.Listen(ListenBacklog);
-                        }
-                        else
-                        {
-                            socket = new Socket((int)_endPoint.FileHandle);
-                        }
-                        ipEndPoint = null;
-                        acceptThread = new AcceptThread(socket);
-                        transportThreads = CreateTransportThreads(ipEndPoint, acceptThread);
+                    case UnixDomainSocketEndPoint unixDomainSocketEndPoint:
+                        var socketPath = unixDomainSocketEndPoint.ToString();
+                        var unixDomainSocket = Socket.Create(AF_UNIX, SOCK_STREAM, 0, blocking: false);
+                        File.Delete(socketPath);
+                        unixDomainSocket.Bind(socketPath);
+                        unixDomainSocket.Listen(ListenBacklog);
+                        acceptThread = new AcceptThread(unixDomainSocket);
+                        _transportThreads = CreateTransportThreads(ipEndPoint: null, acceptThread);
+                        break;
+                    case FileHandleEndPoint fileHandleEndPoint:
+                        var fileHandleSocket = new Socket((int)fileHandleEndPoint.FileHandle);
+                        acceptThread = new AcceptThread(fileHandleSocket);
+                        _transportThreads = CreateTransportThreads(ipEndPoint: null, acceptThread);
                         break;
                     default:
-                        throw new NotSupportedException($"Unknown ListenType: {_endPoint.Type}.");
+                        throw new NotSupportedException($"Unknown ListenType: {EndPoint.GetType()}.");
                 }
 
-                _threads = new ITransportActionHandler[transportThreads.Length + (acceptThread != null ? 1 : 0)];
+                _threads = new ITransportActionHandler[_transportThreads.Length + (acceptThread != null ? 1 : 0)];
                 _threads[0] = acceptThread;
-                for (int i = 0; i < transportThreads.Length; i++)
+                for (int i = 0; i < _transportThreads.Length; i++)
                 {
-                    _threads[i + (acceptThread == null ? 0 : 1)] = transportThreads[i];
+                    _threads[i + (acceptThread == null ? 0 : 1)] = _transportThreads[i];
                 }
 
-                _logger.LogDebug($@"BindAsync {_endPoint}: TC:{_transportOptions.ThreadCount} TA:{_transportOptions.SetThreadAffinity} IC:{_transportOptions.ReceiveOnIncomingCpu} DA:{_transportOptions.DeferAccept}");
+                _logger.LogDebug($@"BindAsync {EndPoint}: TC:{_transportOptions.ThreadCount} TA:{_transportOptions.SetThreadAffinity} IC:{_transportOptions.ReceiveOnIncomingCpu} DA:{_transportOptions.DeferAccept}");
```

Comment:
Isn't this allocating strings even when the log level is higher?

Reply:
Obviously if this is a single call then it doesn't matter.

Reply:
Yeah. Same as before. I don't think bind needs to be too optimized.